crates/core/src/host/wasm_common/module_host_actor.rs

```diff
@@ -531,7 +531,7 @@ impl<T: WasmInstance> WasmModuleInstance<T> {
         };
 
         let tx = tx.unwrap_or_else(|| stdb.begin_mut_tx(IsolationLevel::Serializable));
-        let tx_slot = self.instance.instance_env().tx.clone();
+        let mut tx_slot = self.instance.instance_env().tx.clone();
```

Comment:
Oh I see, we clone the tx slot, so we also need to clone the execution context that it holds.

Reply:
yeah that explains your above comment regarding `Arc<Mutex<>>`